runtime/stdlib/account.go

```diff
@@ -610,7 +610,8 @@ func newAccountKeysAddFunction(
 				panic(err)
 			}
 
-			hashAlgo := NewHashAlgorithmFromValue(inter, locationRange, invocation.Arguments[1])
+			hashAlgoValue := invocation.Arguments[1]
+			hashAlgo := NewHashAlgorithmFromValue(inter, locationRange, hashAlgoValue)
 			weightValue, ok := invocation.Arguments[2].(interpreter.UFix64Value)
```

Comment:
While at it, maybe separate the handling of the arguments a bit more:

```suggestion

			weightValue, ok := invocation.Arguments[2].(interpreter.UFix64Value)
```